dash/include/dash/matrix/internal/Matrix-inl.h

```diff
@@ -80,13 +80,54 @@ ::Matrix(
 
 template <typename T, dim_t NumDim, typename IndexT, class PatternT>
 inline Matrix<T, NumDim, IndexT, PatternT>
+::Matrix(
+  self_t && other)
+: _team(other._team),
+  _size(other._size),
+  _lcapacity(other._lcapacity),
+  _begin(other._begin),
+  _pattern(other._pattern),
+  _glob_mem(other._glob_mem),
+  _lbegin(other._lbegin),
+  _lend(other._lend),
+  _ref(other._ref)
+{
+    // do not free other globmem
+    other._glob_mem = nullptr;
```

Comment:
There are arguments for both sides. IMO nulling the pointers is useful to avoid accidental usage of the dead matrix object. With nulling the user gets an segfault upon first access instead of working on the wrong matrix. I prefer to leave the current implementation.